ownCloud/Viewer/PDFViewerViewController.swift

```diff
@@ -20,44 +20,368 @@ import UIKit
 import ownCloudSDK
 import PDFKit
 
+extension UILabel {
+    func _setupPdfInfoLabel() {
+        self.layer.masksToBounds = true
+        self.layer.cornerRadius = 8.0
+        self.backgroundColor = UIColor.darkGray
+        self.textColor = UIColor.white
+        self.font = UIFont.systemFont(ofSize: 14.0, weight: UIFont.Weight.medium)
+        self.translatesAutoresizingMaskIntoConstraints = false
+        self.textAlignment = .center
+        self.adjustsFontSizeToFitWidth = true
+    }
+}
+
 class PDFViewerViewController: DisplayViewController, DisplayExtension {
 
-	static var customMatcher: OCExtensionCustomContextMatcher?
-	static var displayExtensionIdentifier: String = "org.owncloud.pdfViewer.default"
-	static var supportedMimeTypes: [String]? = ["application/pdf"]
-	static var features: [String : Any]? = [FeatureKeys.canEdit : false]
-
-	override func renderSpecificView() {
-		if let document = PDFDocument(url: source) {
-			let thumbnailsView = PDFThumbnailView(frame: .zero)
-			thumbnailsView.translatesAutoresizingMaskIntoConstraints = false
-			view.addSubview(thumbnailsView)
-			NSLayoutConstraint.activate([
-				thumbnailsView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
-				thumbnailsView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
-				thumbnailsView.leftAnchor.constraint(equalTo: view.leftAnchor),
-				thumbnailsView.widthAnchor.constraint(equalToConstant: 70)
-				])
-
-			thumbnailsView.thumbnailSize = CGSize(width: 70, height: 50)
-
-			let pdfView = PDFView(frame: .zero)
-			pdfView.translatesAutoresizingMaskIntoConstraints = false
-			view.addSubview(pdfView)
-			NSLayoutConstraint.activate([
-				pdfView.topAnchor.constraint(equalTo: view.topAnchor),
-				pdfView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
-				pdfView.leftAnchor.constraint(equalTo: thumbnailsView.rightAnchor),
-				pdfView.rightAnchor.constraint(equalTo: view.rightAnchor)
-				])
-
-			pdfView.document = document
-			pdfView.autoScales = true
-			pdfView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
-			pdfView.displayMode = .singlePageContinuous
-			thumbnailsView.pdfView = pdfView
-		} else {
-			view.backgroundColor = .red
-		}
+    enum ThumbnailViewPosition {
+        case left, right, bottom, none
+        func isVertical() -> Bool {
+            return self == .bottom ? false : true
+        }
+    }
+
+    static let PDFGoToPageNotification = Notification(name: Notification.Name(rawValue: "PDFGoToPageNotification"))
+
+	fileprivate var gotoPageNotificationObserver : Any?
+
+    fileprivate let SEARCH_ANNOTATION_DELAY  = 3.0
+    fileprivate let ANIMATION_DUR = 0.25
+    fileprivate let THUMBNAIL_VIEW_WIDTH_MULTIPLIER: CGFloat = 0.15
+    fileprivate let THUMBNAIL_VIEW_HEIGHT_MULTIPLIER: CGFloat = 0.1
+    fileprivate let FILENAME_CONTAINER_TOP_MARGIN: CGFloat = 10.0
```

Comment:
I've seen that in some places you code constants in different styles, is there some reason to do it?

Reply:
@pablocarmu for me it was not clear if we have any official coding conventions and I didn't find an example of constants in existing code. What is the preferred style?

Reply:
Currently, we don't have any rule defined for this neither any complete style guide apart of what we have in the SwiftLint file.

I think we can discuss the style guide in the new opened issue here https://github.com/owncloud/ios-app/issues/139

Reply:
@pablocarmu fixed